Automattic/wp-calypso
# BackupBrowser: Add backup file browser page scaffolding

This PR is aiming to add the base structure for the backup file browser page, so we could start iterating from here.
This feature is not available for customers yet, so we are using a feature flag to access this page.

Project reference: peaFOp-1oM-p2

## Proposed Changes

* Add new `/backup/:site/contents/:rewindId` route
* Add the structure of the new backups content page that will include the file browser for a specific backup reusing components from [`jetpack/daily-backup-status`](https://github.com/Automattic/wp-calypso/tree/f588238cdb6345a5dbe945fc1abb2f84b0455295/client/components/jetpack/daily-backup-status) such as [`ActionButtons`](https://github.com/Automattic/wp-calypso/blob/f588238cdb6345a5dbe945fc1abb2f84b0455295/client/components/jetpack/daily-backup-status/action-buttons.jsx).
* Add `jetpack/backup-contents-page` feature flag and enable it only for development environment.

## Screenshot
<img width="1139" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1488641/f0699a26-4cd0-46f0-8311-bf5cfb85b45f">

## Testing Instructions
* Checkout this branch
* Start your local environment using `yarn start-jetpack-cloud`
* Select one of your sites with a Jetpack VaultPress Backup plan
* Navigate to VaultPress Backup page
* Pick one date with a backup available (use the calendar and click on any date marked with a dot).
* Copy the link from the `Download Backup` button and replace `download` with `contents` and navigate to that page.
* You should see a page similar to the one shared above.

## Pre-merge Checklist
